frontend/src/Row.js

```diff
@@ -0,0 +1,39 @@
+import {square_value, get_position} from './App'
+import {is_defined_square_value} from './App'
+import React from 'react';
+import Square from './Square'
+
+class Row extends React.Component{
+    constructor(props){
+        super(props);
```

Comment:
i think there's no need for the constructor if no state and no binding functions (but not sure)

Reply:
use function components (instead of classes that inherit from React.Component) wrapped in React.memo() for stateless components that have no life cycle methods.